code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -93,13 +93,13 @@
 		heal_bodypart_damage(1)
 
 /mob/living/simple_animal/hostile/regalrat/AttackingTarget()
-	if (DOING_INTERACTION)
+	if (busy)
 		return
 	. = ..()
 	if(istype(target, /obj/machinery/disposal))
 		src.visible_message("<span class='warning'>[src] starts rummaging through the [target].</span>","<span class='notice'>You rummage through the [target]...</span>")
-		DOING_INTERACTION = TRUE
-		if (do_after(src, 3 SECONDS, target))
```

Comment:
Huh? How come the do_after is gone now?

I'm really confused. All I'm asking you to do is remove `busy` and replace using of it with `DOING_INTERACTION`.

Reply:
Sorry. I was not aware how the whole DOING_INTERACTION worked. Is the new update how it should be looking like?

Reply:
It looked right to me, but you also removed the do_after.